watertap/core/zero_order_base.py

```diff
@@ -178,7 +178,14 @@ def set_recovery_and_removal(self, data, use_default_removal=False):
         Returns:
             None
         """
-        self.set_param_from_data(self.recovery_frac_mass_H2O, data)
+        try:
+            self.set_param_from_data(self.recovery_frac_mass_H2O, data)
+        except KeyError:
+            if self.recovery_frac_mass_H2O[:].fixed:
+                pass
+            else:
+                raise KeyError('fs.unit - database provided does not contain '
+                               'an entry for recovery_frac_mass_H2O for technology.')
```

Comment:
```suggestion
                raise
```
No need to repeat the message here, I think